_drafts/intro-to-working-with-data-in-CUBA.adoc

```diff
@@ -0,0 +1,201 @@
+[source,java]
+----
+@Inject
+DataManager dataManager;
+
+private List<Pet> findEndangeredPets(PetType petType, String city) {
+    return dataManager.load(Pet.class)
+                .query("select e from petclinic$Pet e where e.owner.city = :ownerCity and e.type.id = :petType")
+                .parameter("ownerCity", city)
+                .parameter("petType", petType)
+                .view("pet-with-owner-and-type")
+                .list();
+}
+----
+
+The query part is a little more complicated compared to above, but the same pattern applies. The differences in this example are the usage of the `list()` method as we expect multiple Pets from the database query. Also a specific view has been used, because not only the Pet data, but also the Owner data has to be loaded in order to send out the Mailings.
+
+== Write data with DataManager
+
+DataManager is also responsible for writing data into the database. There are also different ways to achieve it based on the case at hand.
+
+=== Single Item
+
+The first most commonly used case is to update or create a single item in the database. There are no major differences between creating and updating of an entity from the usage perspective.
+
+When revisiting the example from above, where a new Visit for a given Pet ID is created, we only looked at the part which loads the Pet by its Pet ID. In this case, we are more interested in the creation part of the Visit entity. The source code to create and save a Visit looks like this:
+
+
+.VisitServiceBean.java
+[source,java]
+----
+@Inject
+DataManager dataManager;
+
+private Visit createVisitForPet(Pet pet) {
+
+    Visit visit = dataManager.create(Visit.class); // <1>
+
+    visit.setPet(pet);
+    visit.setVisitDate(timeSource.currentTimestamp());
+
+    return dataManager.commit(visit); // <2>
+
+}
+----
+
+<1> an entity instance is created with all initializations in memory
+<2> the entity instance is stored in the DB
+
+The creation of an instance through DataManager does the initialization of the instance as well. It is important to not create an instance via `new Visit()` but rather through this Factory method, so that CUBA can do all the required initialization (like `@PostConstruct` annotated methods e.g.).
+
+Once the entity instance was created, the data of that entity has to be set through its corresponding setter methods. In this case, the database instance of the Pet is set as well as the current date as the visit date.
+
+The process of storing an entity instance contains a few steps like security checks, ID generation, and so on. The return value of the commit method is the stored instance with all properties set that have been added during the storage phase.
+
+TIP: There are a couple over other convenience API methods for storing entites. For more information about those variants see the http://files.cuba-platform.com/javadoc/cuba/{cuba_version}/com/haulmont/cuba/core/global/DataManager.html[DataManager API reference].
```

Comment:
"There is a couple of other..."

Reply:
interesting. In german it would always be "there are". When I read this here - you might be right: https://english.stackexchange.com/questions/50435/there-are-a-couple-of-apples-or-there-is-a-couple-of-apples